sos/sosreport.py

```diff
@@ -740,13 +771,25 @@ def get_temp_file(self):
         return self.tempfile_util.new()
 
     def _set_archive(self):
+        if self.opts.encrypt_key and self.opts.encrypt_pass:
+            self.soslog.error("Password and key-pair encryption requested at "
+                              "the same time. Choose either but not both.")
+            raise SystemExit()
```

Comment:
Could use a mutex group for these, rather than post-hoc validation? I just did this for presets, but I'm not married to it (makes the parser block a bit uglier, but means we don't have to check ourselves).

Reply:
Forgot all about that. Will update.